src/test/java/org/opensearch/neuralsearch/query/NeuralQueryIT.java

```diff
@@ -115,6 +119,49 @@ public void testQueryWithBoostAndImageQuery() {
                 objectToFloat(firstInnerHitMultimodalQuery.get("_score")),
                 DELTA_FOR_SCORE_ASSERTION
             );
+
+            // To save test resources, IT tests for radial search are added below.
+            // Context: https://github.com/opensearch-project/neural-search/pull/697#discussion_r1571549776
+
+            // Test radial search max distance query
+            NeuralQueryBuilder neuralQueryWithMaxDistanceBuilder = new NeuralQueryBuilder(
+                TEST_KNN_VECTOR_FIELD_NAME_1,
+                TEST_QUERY_TEXT,
+                "",
+                modelId,
+                null,
+                100.0f,
+                null,
+                null,
+                null
+            );
+
+            Map<String, Object> searchResponseAsMap = search(TEST_BASIC_INDEX_NAME, neuralQueryWithMaxDistanceBuilder, 1);
+            Map<String, Object> firstInnerHit = getFirstInnerHit(searchResponseAsMap);
+
+            assertEquals("1", firstInnerHit.get("_id"));
+            float expectedScoreWithMaxDistanceQuery = computeExpectedScore(modelId, testVector, TEST_SPACE_TYPE, TEST_QUERY_TEXT);
+            assertEquals(expectedScoreWithMaxDistanceQuery, objectToFloat(firstInnerHit.get("_score")), DELTA_FOR_SCORE_ASSERTION);
+
+            // Test radial search min score query
+            NeuralQueryBuilder neuralQueryWithMinScoreBuilder = new NeuralQueryBuilder(
+                TEST_KNN_VECTOR_FIELD_NAME_1,
+                TEST_QUERY_TEXT,
+                "",
+                modelId,
+                null,
+                null,
+                0.01f,
+                null,
+                null
+            );
+
+            searchResponseAsMap = search(TEST_BASIC_INDEX_NAME, neuralQueryWithMinScoreBuilder, 1);
+            firstInnerHit = getFirstInnerHit(searchResponseAsMap);
```

Comment:
Can you please create separate variable for each scenario instead of reusing same vars each time? It makes code more maintainable

Reply:
Yes, updated.